File: merge_sort.js
const mergeSort = (array) => {
    if(array.length < 2) {
        return array;
    }
    const length = array.length;
    const left = array.slice(0, Math.floor(length/2));
    const right = array.slice(Math.floor(length/2));
    return merge(mergeSort(left), mergeSort(right));
}

const merge = (left, right) => {
    const sorted = [];
    while(left.length > 0 && right.length > 0) {
        if(left[0] < right[0]) {
            sorted.push(left.shift());
        } else {
            sorted.push(right.shift());
        }
    }
    return sorted.concat(left, right);
}

File: merge_sort.py
def merge_sort(unsorted_list):
    if len(unsorted_list) < 2:
        return unsorted_list
    middle = int(len(unsorted_list)/2)
    left = unsorted_list[:middle]
    right = unsorted_list[middle:]
    return merge(merge_sort(left), merge_sort(right))


def merge(left, right):
    sorted = []
    while len(left) > 0 and len(right) > 0:
        if left[0] > right[0]:
            sorted.append(right.pop(0))
        else:
            sorted.append(left.pop(0))
    return sorted + left + right
